test/e2e-helm/e2e_helm_project_on_cluster_test.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e_helm_test
+
+import (
+	"encoding/base64"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo" //nolint:golint
+	. "github.com/onsi/gomega" //nolint:golint
+	kbtestutils "sigs.k8s.io/kubebuilder/test/e2e/utils"
+
+	testutils "github.com/operator-framework/operator-sdk/test/internal"
+)
+
+var _ = Describe("Running Helm projects", func() {
+	var controllerPodName string
+	const curlImage = "curlimages/curl:latest"
+
+	Context("built with operator-sdk", func() {
+		BeforeEach(func() {
+			By("Ensuring that the namespace is not created")
+			verifyControllerUp := func() error {
+				_, err := tc.Kubectl.Command("get", "namespace", tc.Kubectl.Namespace)
+				if strings.Contains(err.Error(), "(NotFound): namespaces") {
+					return err
+				}
+				return nil
+			}
+			Eventually(verifyControllerUp, time.Minute, time.Second).ShouldNot(Succeed())
```

Comment:
It seems like this would be better suited as the last check in `AfterEach`. Logically, we're calling `make undeploy` and then waiting for the namespace to be cleaned up.

It seems extremely unlikely that the namespace will exist before the test starts because its name is derived from a random string.

Reply:
Hi @joelanford, 

We scaffold the project and all tests are done in the suite will use the same project. So, if one test deploy and undeploy another one will deploy and undeploy the same project in the same namespace. Because of this, the validation is required.  But we can move it for the After. as part of the cleanup.

Reply:
Yeah, I agree that we need this check. Just seems like it fits more in `AfterEach`.